### ISSUE TITLE
nexus-inference: GbdtF32 — gradient-boosted decision tree inference (I4)

### ISSUE THREAD

**Issue by Abso1ut3Zer0** [Abso1ut3Zer0/nexus#308]
## Summary

Gradient-boosted decision tree inference. Flat array-of-nodes representation for cache-friendly traversal. **Probably the highest-utility single inference type** — tabular features + LightGBM/XGBoost is the dominant pattern in production trading ML.

**Crate:** `nexus-inference`
**Module:** `nexus_inference::gbdt::GbdtF32`

## API sketch

```rust
pub struct GbdtF32<const MAX_TREES: usize, const MAX_DEPTH: usize, const FEATURES: usize> {
    trees: [TreeNodes; MAX_TREES],
    n_trees: usize,
    learning_rate: f32,
    base_score: f32,
}

struct TreeNodes {
    feature_idx: [u16; NODES],
    threshold: [f32; NODES],
    left_child: [u16; NODES],
    right_child: [u16; NODES],
    leaf_value: [f32; NODES],
    is_leaf: [bool; NODES],
}

impl<...> GbdtF32<...> {
    pub fn predict(&self, features: &[f32; FEATURES]) -> f32;
    pub fn predict_trees(&self, features: &[f32; FEATURES], n_trees: usize) -> f32;
}
```

## Latency targets

| Trees | Depth | Target |
|---|---|---|
| 50 | 6 | 500 ns |
| 100 | 6 | 1 μs |
| 200 | 8 | 3 μs |
| 500 | 8 | 8 μs |

## Design constraints

- Flat array-of-nodes per tree — cache-friendly sequential access
- Breadth-first vs depth-first layout TBD (benchmark before committing)
- `predict_trees(n)` for early-stopping / partial ensemble
- Categorical splits via bitmask
- Missing value routing (NaN → default direction, like LightGBM)
- Multi-class via softmax over K output trees
- `no_std` compatible (needs `alloc` for variable tree count)
- Pair with I6 weight loaders (LightGBM, XGBoost formats)

## Build order

**First wave** — highest priority. Build first.

*From nexus-inference roadmap item I4.*

**Comment by Abso1ut3Zer0** [comment]
Shipped in #343. GbdtF64/GbdtF32 with predict/predict_unchecked/predict_n/predict_n_unchecked. False-branch-next layout, flat storage, ~225ns for 50x6 unchecked.
